test/alphabet/gap/gapped_alphabet_test.cpp

```diff
@@ -43,74 +41,186 @@
 
 using namespace seqan3;
 
-TEST(gapped_alphabet_test, test_alphabet_concept)
+TEST(gapped_alphabet_test, default_constructor)
 {
-    EXPECT_TRUE(alphabet_concept<gapped_alphabet<dna4>>);
+    using alphabet_t = gapped_alphabet<dna4>;
+
+    constexpr auto letter1 = alphabet_t{};
+    EXPECT_EQ(letter1._value, 0);
 }
 
-TEST(gapped_alphabet_test, test_implicit_inner_type_compatibility)
+TEST(gapped_alphabet_test, base_alphabet_copy_constructor)
 {
-    EXPECT_EQ(gapped_alphabet<dna4>{0}, gapped_alphabet<dna4>{}.assign_rank(0));
+    using alphabet_t = gapped_alphabet<dna4>;
+    using variant_t = alphabet_t::variant_type;
+
+    constexpr alphabet_t letter0{dna4::A};
+    constexpr alphabet_t letter1 = {dna4::C};
+    constexpr alphabet_t letter2 = static_cast<variant_t>(dna4::G);
+    constexpr alphabet_t letter3{dna4::T}; // letter3 = dna4::T; does not work
+
+    alphabet_t letter4{dna4::A};
+    alphabet_t letter5 = {dna4::C};
+    alphabet_t letter6 = static_cast<variant_t>(dna4::G);
+    alphabet_t letter7{dna4::T}; // letter7 = dna4::T; does not work
+
+    constexpr alphabet_t letter8{gap::GAP}; // letter3 = dna4::T; does not work
+    alphabet_t letter9{gap::GAP};
+
+    EXPECT_EQ(letter0.to_rank(), 0);
```

Comment:
In this case it is first testing the basic functionality to initialize it from the given alphabets.

Because there is no other readable way to construct this class.